src/scene_teleport.cpp

```diff
@@ -42,7 +43,7 @@ void Scene_Teleport::Update() {
 	teleport_window->Update();
 
 	if (Input::IsTriggered(Input::DECISION)) {
-		Game_System::SePlay(Game_System::GetSystemSE(Game_System::SFX_UseItem));
+		Game_System::SePlay(skill->sound_effect);
```

Comment:
Somehow my comment disappeared...?

This causes a crash now. The `skill` pointer here is null when a teleport item is used. We need it in the else block and there we have to lookup the skill from the item. 

The code is confusing as is. I think it would make more sense if `skill` pointer was always valid and `item` was the optional piece.